cccc/scripts/filter_hf_ds.py

```diff
@@ -0,0 +1,179 @@
+import argparse
+import gzip
+import json
+import logging
+import os
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from functools import partial
+from pathlib import Path
+from urllib.parse import urlparse
+
+from tqdm import tqdm
+
+# Filters the data to only include the URLs in the urls.txt file
+# Requires a local copy of https://huggingface.co/datasets/allenai/dolma-cccc
+# keeps the same folder structure in the destination directory
+
+
+def setup_logging(log_level: str) -> None:
+    """
+    Configures the logging settings.
+    """
+    numeric_level = getattr(logging, log_level.upper(), None)
+    if not isinstance(numeric_level, int):
+        raise ValueError(f"Invalid log level: {log_level}")
+    logging.basicConfig(
+        level=numeric_level,
+        format="%(asctime)s [%(levelname)s] %(processName)s %(message)s",
+        handlers=[logging.StreamHandler()],
+    )
+
+
+def read_file_to_list(file_path):
+    with open(file_path, "r") as file:
+        return [line.strip() for line in file]
+
+
+def process_file(
+    file_path: str, source_dir: str, dest_dir: str, urls: set
+) -> bool | None:
+    """
+    Processes a single json.gz file by filtering its contents and saving it to the destination directory.
+    """
+    file_path = Path(file_path)
+    source_dir = Path(source_dir)
+    dest_dir = Path(dest_dir)
+
+    # Compute the relative path to maintain directory structure
+    rel_path = file_path.relative_to(source_dir)
+    dest_file_path = dest_dir / rel_path
+
+    # Ensure the destination directory exists
+    dest_file_path.parent.mkdir(parents=True, exist_ok=True)
+
+    # Open the source and destination files
+    try:
+        with gzip.open(file_path, "rt", encoding="utf-8") as f_in, gzip.open(
+            dest_file_path, "wt", encoding="utf-8"
+        ) as f_out:
+            for line in f_in:
+                try:
+                    item = json.loads(line)
+                    if filter_condition(item, urls):
+                        json.dump(item, f_out)
+                        f_out.write("\n")
+                except json.JSONDecodeError as e:
+                    logging.error(f"Error decoding JSON in file {file_path}: {e}")
+    except Exception as e:
+        logging.error(f"Error processing file {file_path}: {e}")
+        return False
+    else:
+        logging.info(f"Successfully processed file: {file_path}")
+        return True
+
+
+def extract_suburls(url: str) -> list:
+    parsed_url = urlparse(url)
+    domain = parsed_url.netloc
+    path_parts = parsed_url.path.strip("/").split("/")
+
+    suburls = [
+        domain,
+    ]
+    current_path = domain
+
+    for part in path_parts[:-1]:
+        current_path = f"{current_path}/{part}"
+        suburls.append(current_path)
+
+    return suburls
+
+
+def filter_condition(item: dict, URLS: set) -> bool:
```

Comment:
All uppercase should be used for global variables. I like that this is passed into the function, but it should be called `urls`